packages/scandipwa/src/route/ProductPage/ProductPage.container.js

```diff
@@ -167,10 +169,23 @@ export class ProductPageContainer extends PureComponent {
 
         const configurableVariantIndex = getVariantIndex(variants, parameters);
 
+        const newOptionsData = options.reduce((acc, { option_id, required }) => {
+            if (required) {
+                acc.push(option_id);
+            }
+
+            return acc;
+        }, []);
+
+        const prevOptions = productOptionsData?.requiredOptions || [];
+        const requiredOptions = [...prevOptions, ...newOptionsData];
+
         return {
             parameters,
             currentProductSKU,
-            configurableVariantIndex
+            configurableVariantIndex,
+            productOptionsData:
+                { ...productOptionsData, requiredOptions }
```

Comment:
Please comment on this code. Also, is it used somewhere? I see that here the additional information is provided, but I don't see any additional usages. Is this intentional?

Reply:
@yeegor 

These props have already been referred to before my update:
https://github.com/scandipwa/scandipwa/blob/master/packages/scandipwa/src/component/ProductActions/ProductActions.component.js#L350
```
<AddToCart
              product={ product }
              configurableVariantIndex={ configurableVariantIndex }
              mix={ { block: 'ProductActions', elem: 'AddToCart' } }
              quantity={ quantity }
              groupedProductQuantity={ groupedProductQuantity }
              onProductValidationError={ onProductValidationError }
              productOptionsData={ productOptionsData }
            />
```
But they haven't been updated correctly for the specific case described in https://github.com/scandipwa/scandipwa/issues/2057

So there is no need for additional usages. You just have to fix the value you are referring to in order to access the correct value in `AddToCart` component.

And please note the method I have updated is `getDerivedStateFromProps`. `productOptionsData` have been in the `state` before:
https://github.com/scandipwa/scandipwa/blob/86400e6328c3827052f4749c8827bd7a7cf607bc/packages/scandipwa/src/route/ProductPage/ProductPage.container.js#L88